### DISCUSSION TITLE
Tree-Shake python dependencies

### DISCUSSION THREAD

**Discussion by baseplate-admin** [kivy/kivy-github-discussions#22] [Ideas]
As it stands,

The apk generated by `python-for-android` are large in size and nature. 

I was wondering if it was possible to tree shake the android venv to reduce size.

Possible paths :
* **Compile python with bare minimum**: This means we remove the `stdlib`  to a bare minimum for our app to function.
* **Remove codes from venv** : Tools like [`dead`](https://github.com/asottile/dead) and [`vulture`](https://github.com/jendrikseipp/vulture) can be of use here.
* **Use better compression algorithm** : Maybe support `brotli`/`zstd` instead of `.zip` ( or maybe even `7-zip` )

**Comment by baseplate-admin** [comment]
Linked issue in flet : https://github.com/flet-dev/flet/discussions/2319#discussion-6032729

**Comment by baseplate-admin** [comment]
Hi @Julian-O, can we move this to a [issue](https://github.com/kivy/python-for-android/issues)

**Reply by Julian-O** [reply]
I don't have any fancy button to do that. Would you like to cut-and-paste into one?

**Reply by baseplate-admin** [reply]
https://github.com/kivy/kivy-github-discussions/issues/26#issue-2098619036 

Could you please transfer this issue to the python for android repo?

**Reply by baseplate-admin** [reply]
![image](https://github.com/kivy/kivy-github-discussions/assets/61817579/19b1fb2c-af43-431c-9cda-71fb9238c617)

See the button here? ( create issue from discussion? ). This is what i wanted to do.

After you create an issue, you can move it to another repository ( as a moderator )

**Reply by Julian-O** [reply]
I see that option now, thanks, but I don't have sufficient access to make it happen. @misl6, can you please help?
